### PR TITLE
Fix HLS OpenAPI warning

### PR DESCRIPTION
**Pull request by Rados13**
## Acknowledging the stipulations set forth:
 - [x] I hereby confirm that a Pull Request involving updates to the Software Development Kit (SDK) has been smoothly merged, currently awaits processing, or is otherwise deemed unnecessary in this context.
 - [x] I also affirm that another Pull Request, specifically addressing updates to the documentation body (commonly referred to as 'docs'), has either been successfully incorporated, is in the process of review, or is considered superfluous under the prevailing circumstances.

OpenAPI warning is fixed by how object `S3Credentials` is referenced. 
Error with AuthCache was fixed by providing `AuthConfigAdapter`, which is the way `AuthCache` is tested in `ExAws`.

### CODE REVIEW

File: lib/fishjam_web/api_spec/component/hls.ex [hunk 1]
```diff
@@ -100,12 +70,7 @@ defmodule FishjamWeb.ApiSpec.Component.HLS do
           description: "Whether the video is stored after end of stream",
           default: false
         },
-        s3: %Schema{
-          type: :object,
-          description: "Credentials to AWS S3 bucket.",
-          oneOf: [S3],
-          nullable: true
-        },
+        s3: S3.schema(),
```
Comment (sgfn) [inline lib/fishjam_web/api_spec/component/hls.ex:73]: This isn't the same as before -- the field is no longer nullable. Why this change?

File: test/fishjam_web/integration/peer_socket_test.exs [hunk 2]
```diff
@@ -178,13 +179,26 @@ defmodule FishjamWeb.Integration.PeerSocketTest do
...
+               for {k, v} <- metrics_after_one_tick do
+                 assert Map.fetch!(metrics_to_compare, k) == v
+               end
+
+               {:halt, true}
+             else
+               {:cont, false}
+             end
+           end),
+           "Metrics isn't present after 1,5 second"
```
Comment (sgfn) [inline test/fishjam_web/integration/peer_socket_test.exs:201]:
```suggestion
           "Metrics aren't present after 1.5 seconds"
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ orbs:
 executors:
  machine_executor_amd64:
    machine:
-     image: ubuntu-2204:2022.04.2
+     image: default
```
Comment (sgfn) [inline .circleci/config.yml:9]: Was there an issue with locking to a specific image?

File: lib/fishjam/component/hls/manager.ex [hunk 4]
```diff
@@ -55,6 +55,8 @@ defmodule Fishjam.Component.HLS.Manager do
     unless is_nil(hls_options.s3), do: upload_to_s3(hls_dir, room_id, hls_options.s3)
     unless hls_options.persistent, do: remove_hls(hls_dir, room_id)
 
+    Logger.info("Enginer is down and HLS manager finished uploading stream")
```
Comment (sgfn) [inline lib/fishjam/component/hls/manager.ex:58]:
```suggestion
    Logger.info("Engine is down and HLS manager finished uploading stream")
```

File: config/config.exs [hunk 5]
```diff
@@ -29,12 +31,12 @@ config :phoenix, :logger, false
...
-    [application: :membrane_rtc_engine_hls, level_lower_than: :warning],
-    [application: :membrane_rtc_engine_rtsp, level_lower_than: :warning],
-    [application: :membrane_rtc_engine_file, level_lower_than: :warning],
-    [application: :membrane_rtc_engine_sip, level_lower_than: :warning]
+    # [application: :membrane_rtc_engine, level_lower_than: :warning],
+    # [application: :membrane_rtc_engine_webrtc, level_lower_than: :warning],
+    # [application: :membrane_rtc_engine_hls, level_lower_than: :warning],
+    # [application: :membrane_rtc_engine_rtsp, level_lower_than: :warning],
+    # [application: :membrane_rtc_engine_file, level_lower_than: :warning],
+    # [application: :membrane_rtc_engine_sip, level_lower_than: :warning]
```
Comment (sgfn) [inline config/config.exs:39]: Leftover?

File: mix.exs [hunk 6]
```diff
@@ -11,6 +11,7 @@ defmodule Fishjam.MixProject do
       aliases: aliases(),
       deps: deps(),
       dialyzer: dialyzer(),
+      consolidate_protocols: false,
```
Comment (sgfn) [inline mix.exs:14]: Why this change?

File: .circleci/config.yml (hunk 3 above)
Comment (Rados13) [reply .circleci/config.yml:9]: Yup, circleCI deprecated this specific version, and the suggested version by them is default.
[Post about this deprecation.](https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177)

File: mix.exs (hunk 6 above)
Comment (sgfn) [reply mix.exs:14]: Can we add a `# TODO: remove once [...]` comment here?

File: test/fishjam/component/hls/manager_test.exs [hunk 7]
```diff
@@ -37,29 +39,34 @@ defmodule Fishjam.Component.HLS.ManagerTest do
     hls_dir: hls_dir,
     options: options
   } do
-    MockManager.http_mock_expect(0, status_code: 200)
+    Application.put_env(:ex_aws, :awscli_auth_adapter, Fishjam.Adapter)
```
Comment (Karolk99) [inline test/fishjam/component/hls/manager_test.exs:42]: why do we need this envs?

File: test/fishjam/component/hls/manager_test.exs [hunk 8]
```diff
@@ -37,29 +39,34 @@ defmodule Fishjam.Component.HLS.ManagerTest do
...
+
     pid = MockManager.start_mock_engine()
 
     {:ok, manager} = Manager.start(room_id, pid, hls_dir, options)
     ref = Process.monitor(manager)
 
+    assert Process.alive?(pid)
```
Comment (Karolk99) [inline test/fishjam/component/hls/manager_test.exs:49]: I don't think this is necessary, as next line does the same thing

File: test/fishjam_web/integration/peer_socket_test.exs [hunk 9]
```diff
@@ -24,9 +24,10 @@ defmodule FishjamWeb.Integration.PeerSocketTest do
 
     alias FishjamWeb.PeerSocket
 
-    socket "/socket/peer", PeerSocket,
+    socket("/socket/peer", PeerSocket,
```
Comment (Karolk99) [inline test/fishjam_web/integration/peer_socket_test.exs:27]: 👎

**Review by Karolk99 — APPROVED** [review]

**Review by sgfn — APPROVED** [review]
